arm64 windows and arm64 uwp builds are broken

See the failed azure pipelines in this [PR](https://github.com/microsoft/vcpkg/pull/40911/checks?check_run_id=30014623379) and attached logs. This prevents from bumping the version to 0.3.28 in the vcpkg port.

```
Change Dir: 'D:/b/openblas/arm64-windows-dbg'

Run Build Command(s): "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j33 install
[1/1740] C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\arm64\cl.exe    -ID:\b\openblas\src\v0.3.28-8b4f07f037.clean -ID:\b\openblas\arm64-windows-dbg /nologo /DWIN32 /D_WINDOWS /utf-8 /MP   -DGEMM_GEMV_FORWARD -DNO_LAPACK -DNO_LAPACKE -DNO_WARMUP -DMAX_CPU_NUMBER=0 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.28\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fointerface\CMakeFiles\interface.dir\CMakeFiles\cnrm2.c.obj /Fdinterface\CMakeFiles\interface.dir\ /FS -c D:\b\openblas\arm64-windows-dbg\interface\CMakeFiles\cnrm2.c
FAILED: interface/CMakeFiles/interface.dir/CMakeFiles/cnrm2.c.obj 
C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\arm64\cl.exe    -ID:\b\openblas\src\v0.3.28-8b4f07f037.clean -ID:\b\openblas\arm64-windows-dbg /nologo /DWIN32 /D_WINDOWS /utf-8 /MP   -DGEMM_GEMV_FORWARD -DNO_LAPACK -DNO_LAPACKE -DNO_WARMUP -DMAX_CPU_NUMBER=0 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.28\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fointerface\CMakeFiles\interface.dir\CMakeFiles\cnrm2.c.obj /Fdinterface\CMakeFiles\interface.dir\ /FS -c D:\b\openblas\arm64-windows-dbg\interface\CMakeFiles\cnrm2.c
D:\b\openblas\src\v0.3.28-8b4f07f037.clean\common_arm64.h(60): error C2065: '__asm__': undeclared identifier
D:\b\openblas\src\v0.3.28-8b4f07f037.clean\common_arm64.h(60): error C2143: syntax error: missing ';' before 'volatile'
[2/1740] C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\arm64\cl.exe    -ID:\b\openblas\src\v0.3.28-8b4f07f037.clean -ID:\b\openblas\arm64-windows-dbg /nologo /DWIN32 /D_WINDOWS /utf-8 /MP   -DGEMM_GEMV_FORWARD -DNO_LAPACK -DNO_LAPACKE -DNO_WARMUP -DMAX_CPU_NUMBER=0 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.28\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fointerface\CMakeFiles\interface.dir\CMakeFiles\sasum.c.obj /Fdinterface\CMakeFiles\interface.dir\ /FS -c D:\b\openblas\arm64-windows-dbg\interface\CMakeFiles\sasum.c
FAILED: interface/CMakeFiles/interface.dir/CMakeFiles/sasum.c.obj 
```

[install-arm64-windows-dbg-out.log](https://github.com/user-attachments/files/17309442/install-arm64-windows-dbg-out.log)
[install-arm64-uwp-dbg-out.log](https://github.com/user-attachments/files/17309618/install-arm64-uwp-dbg-out.log)

--- a/common_arm64.h
+++ b/common_arm64.h
@@ -55,6 +55,7 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef ASSEMBLER
 
 
+#ifndef NO_AFFINITY
 static __inline int WhereAmI(void){
   uint64_t ret;
   __asm__ volatile (
@@ -67,6 +68,7 @@ static __inline int WhereAmI(void){
   if ((int)ret <0) ret = 0;
   return (int)ret;
 }
+#endif
 
 static __inline void blas_lock(volatile BLASULONG *address){
 

